weave

```diff
@@ -909,24 +909,14 @@ ipam_cidrs() {
     shift 1
     ALL_CIDRS=""
     IPAM_CIDRs=""
-    if [ $# -eq 0 ] ; then
-        # If no addresses passed in, get one in the default subnet
-        IPAM_CIDRS=$(call_weave POST /ip/$CONTAINER_ID$CHECK_ALIVE) || return 1
-        if [ "$IPAM_CIDRS" = "404 page not found" ] ; then
-            echo "No IP address supplied and IP address allocation is disabled" >&2
-            return 1
-        elif ! is_cidr "$IPAM_CIDRS" ; then
-            echo "$IPAM_CIDRS" >&2
-            return 1
-        fi
-        ALL_CIDRS="$IPAM_CIDRS"
-    fi
-    while [ $# -gt 0 ] ; do
-        if [ "${1%:*}" = "net" ] ; then
-            if [ "$1" = "net:default" ] ; then
+    # If no addresses passed in, get one in the default subnet
+    [ $# -eq 0 ] && set -- net:default
```

Comment:
I tend to prefer writing these things s.t. the whole expression always succeeds, i.e. `[ $# -gt 0 ] || set -- net:default`.